src/inference/src/ie_core.cpp

```diff
@@ -505,12 +518,156 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
+                auto res =
+                    GetConfig(deviceNameWithoutBatch, CONFIG_KEY(PERFORMANCE_HINT_NUM_REQUESTS)).as<std::string>();
+                requests = PerfHintsConfig::CheckPerformanceHintRequestValue(res);
+                const auto& reqs = config_with_batch.find(CONFIG_KEY(PERFORMANCE_HINT_NUM_REQUESTS));
+                if (reqs != config_with_batch.end())
+                    requests =
+                        static_cast<unsigned int>(PerfHintsConfig::CheckPerformanceHintRequestValue(reqs->second));
+                if (requests) {
+                    optimalBatchSize = std::max(1u, std::min(requests, optimalBatchSize));
+                }
+            }
+            if (optimalBatchSize > 1 || !deviceNameWithBatchSize.empty()) {
+                auto function = network.getFunction();
+                // have to execute the DetectionOutput separately (without batching)
+                // as this layer mix-in the values from the different inputs (batch id)
+                bool bDetectionOutput = false;
+                for (auto&& node : function->get_ops()) {
+                    auto isDetectionOutputParent = [](decltype(node)& nd) {
+                        for (size_t n = 0; n < nd->get_input_size(); n++) {
+                            // the code below doesn't need to separate the versions (opsets) of the DetectionOutput
+                            // so type_info name check is enough
+                            // (if in a future there will be a new ver that doesn't mix the batch, this will be new op)
+                            if (!std::strcmp("DetectionOutput", nd->get_input_node_ptr(n)->get_type_info().name))
+                                return true;
+                        }
+                        return false;
+                    };
+
+                    if (!std::strcmp("DetectionOutput", node->get_type_info().name) ||
+                        (!std::strcmp("Result", node->get_type_info().name) && isDetectionOutputParent(node))) {
+                        node->get_rt_info()["affinity"] = deviceNameWithoutBatch;
+                        bDetectionOutput = true;
+                    } else {
+                        node->get_rt_info()["affinity"] = "BATCH";
+                    }
+                }
+                auto batchConfig = deviceNameWithBatchSize.empty()
+                                       ? deviceNameWithoutBatch + "(" + std::to_string(optimalBatchSize) + ")"
+                                       : deviceNameWithBatchSize;
+                if (bDetectionOutput) {
+                    deviceName = "HETERO:BATCH," + deviceNameWithoutBatch;
+                    config_with_batch[CONFIG_KEY(AUTO_BATCH)] = batchConfig;
+                } else {
+                    deviceName = "BATCH:" + batchConfig;
+                }
+            }
+        } catch (std::exception& e) {
+            // No Auto-Batching enabled
+        }
+    }
```

Comment:
can we move this function outside of the `ie_core.cpp`. It's 150 lines of bull shit which is not Core related

Reply:
Move to plugin load network itself. You can load and return the network from any plugin even inside the auto batch plugin as you have access to ICore API inside the plugin.

Reply:
will completely refactor later, now simply remove to disable the Auto-batching as being on by default in the product